provider/openstack/networking.go

```diff
@@ -228,18 +237,20 @@ func getExternalNeutronNetworksByAZ(e *Environ, azName string) ([]string, error)
 		}
 	}
 	// Find all external networks in availability zone
-	networks, err := neutron.ListNetworksV2()
+	networks, err := neutron.ListNetworksV2(externalNetworkFilter())
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	netIds := make([]string, 0)
 	for _, network := range networks {
-		if network.External == true {
-			for _, netAZ := range network.AvailabilityZones {
-				if azName == netAZ {
-					netIds = append(netIds, network.Id)
-					break
-				}
+		// TODO (hml): OpenStack Compute and Network AZs have no direct relation,
```

Comment:
It doesn't appear so, which presents a problem.  Fixing this area of code is going to be messy, so I wanted to get the note in.